tests/lax_numpy_reducers_test.py

```diff
@@ -866,5 +867,28 @@ def testCumulativeSumBool(self):
                              dtype=jnp.bool_)
     np.testing.assert_array_equal(np.array([[True], [True], [False]]), out)
 
+  def test__promote_integer_dtype(self) -> None:
+    self.assertTrue(_promote_integer_dtype(jnp.int4), dtypes.int_)
+    self.assertTrue(_promote_integer_dtype(jnp.uint4), dtypes.int_)
+    self.assertTrue(_promote_integer_dtype(jnp.int2), dtypes.int_)
+    self.assertTrue(_promote_integer_dtype(jnp.uint2), dtypes.int_)
+
+  @jtu.sample_product(
+    [dtype for dtype in int_dtypes + unsigned_dtypes],
+  )
+  @jtu.ignore_warning(category=NumpyComplexWarning)
+  def test_low_bitwidth_sum(self, dtype) -> None:
+    sum_jit = jax.jit(jnp.sum)
+    arr = jax.random.randint(
+        key=jax.random.PRNGKey(0),
+        shape=(10, 10),
+        minval=0,
```

Comment:
We should use `jtu.rand_int()` here because it gives us more coverage than relying on hard-coded seeds like `key(0)`.

Reply:
I switched for the `rng_factory` like in other tests is this correct?